### ISSUE TITLE
Training process stop

### ISSUE THREAD

**Issue by liaozhihao2757** [SenseTime-FVG/OpenDWM#20]
ShardingStrategy._HYBRID_SHARD_ZERO2 since the world size is 1.
  warnings.warn(
The pipeline is loaded.
The training dataset is loaded with 165280 items.
The validation dataset is loaded with 2082 items.
Step 100 (1.1 s/step), loss: 0.4138
Step 200 (1.1 s/step), loss: 0.4006
Step 300 (1.1 s/step), loss: 0.3869
Step 400 (1.1 s/step), loss: 0.3922
Step 500 (1.1 s/step), loss: 0.3970
Step 600 (1.1 s/step), loss: 0.3972
Step 700 (1.1 s/step), loss: 0.3860
Step 800 (1.1 s/step), loss: 0.3964
Step 900 (1.1 s/step), loss: 0.3956
Step 1000 (1.1 s/step), loss: 0.3951
Step 1100 (1.1 s/step), loss: 0.4040
Step 1200 (1.1 s/step), loss: 0.4008
Step 1300 (1.1 s/step), loss: 0.4023
Step 1400 (1.1 s/step), loss: 0.3855
Step 1500 (1.1 s/step), loss: 0.3996
Step 1600 (1.1 s/step), loss: 0.3830
Step 1700 (1.1 s/step), loss: 0.3899
Step 1800 (1.1 s/step), loss: 0.3845
Step 1900 (1.1 s/step), loss: 0.3850
Step 2000 (1.1 s/step), loss: 0.3860
Step 2100 (1.1 s/step), loss: 0.3933
Step 2200 (1.1 s/step), loss: 0.3967
Step 2300 (1.1 s/step), loss: 0.3901
Step 2400 (1.1 s/step), loss: 0.3866
Step 2500 (1.1 s/step), loss: 0.3867
Step 2600 (1.1 s/step), loss: 0.3852
Step 2700 (1.1 s/step), loss: 0.3885
Step 2800 (1.1 s/step), loss: 0.3861
Step 2900 (1.1 s/step), loss: 0.4007
Step 3000 (1.1 s/step), loss: 0.3900
Step 3100 (1.1 s/step), loss: 0.3929
Step 3200 (1.1 s/step), loss: 0.3968
Step 3300 (1.1 s/step), loss: 0.3865
Step 3400 (1.1 s/step), loss: 0.3889
Step 3500 (1.1 s/step), loss: 0.3895
Step 3600 (1.1 s/step), loss: 0.3879
Step 3700 (1.1 s/step), loss: 0.3901
Step 3800 (1.1 s/step), loss: 0.3822
Step 3900 (1.1 s/step), loss: 0.3844
Step 4000 (1.1 s/step), loss: 0.3819
Step 4100 (1.1 s/step), loss: 0.3938
Step 4200 (1.1 s/step), loss: 0.3870
Step 4300 (1.1 s/step), loss: 0.3857
Step 4400 (1.1 s/step), loss: 0.3877
Step 4500 (1.1 s/step), loss: 0.4036
Step 4600 (1.1 s/step), loss: 0.3942
Step 4700 (1.1 s/step), loss: 0.3904
Step 4800 (1.1 s/step), loss: 0.3868
Step 4900 (1.1 s/step), loss: 0.3802
Step 5000 (1.1 s/step), loss: 0.3790
Step 5100 (1.1 s/step), loss: 0.3817
Step 5200 (1.1 s/step), loss: 0.3917
Step 5300 (1.1 s/step), loss: 0.3880
Step 5400 (1.1 s/step), loss: 0.3836
Step 5500 (1.1 s/step), loss: 0.3814
Step 5600 (1.1 s/step), loss: 0.3830
Step 5700 (1.1 s/step), loss: 0.3789
Step 5800 (1.1 s/step), loss: 0.3794
Step 5900 (1.1 s/step), loss: 0.3862
Step 6000 (1.1 s/step), loss: 0.3864
Step 6100 (1.1 s/step), loss: 0.3983
Step 6200 (1.1 s/step), loss: 0.3938
Step 6300 (1.1 s/step), loss: 0.3964
Step 6400 (1.1 s/step), loss: 0.3926
Step 6500 (1.1 s/step), loss: 0.3880
Step 6600 (1.1 s/step), loss: 0.3896
Step 6700 (1.1 s/step), loss: 0.3783
Step 6800 (1.1 s/step), loss: 0.3886
Step 6900 (1.1 s/step), loss: 0.3884
Step 7000 (1.1 s/step), loss: 0.3858
Step 7100 (1.1 s/step), loss: 0.3831
Step 7200 (1.1 s/step), loss: 0.3873
Step 7300 (1.1 s/step), loss: 0.3899
Step 7400 (1.1 s/step), loss: 0.3841
Step 7500 (1.1 s/step), loss: 0.3813
Step 7600 (1.1 s/step), loss: 0.3855
Step 7700 (1.1 s/step), loss: 0.3804
Step 7800 (1.1 s/step), loss: 0.3756
Step 7900 (1.1 s/step), loss: 0.3787
Step 8000 (1.1 s/step), loss: 0.3934
Step 8100 (1.1 s/step), loss: 0.3790
Step 8200 (1.1 s/step), loss: 0.3869
Step 8300 (1.1 s/step), loss: 0.3852
Step 8400 (1.1 s/step), loss: 0.3802
Step 8500 (1.1 s/step), loss: 0.3855
Step 8600 (1.1 s/step), loss: 0.3785
Step 8700 (1.1 s/step), loss: 0.3855
Step 8800 (1.1 s/step), loss: 0.3815
Step 8900 (1.1 s/step), loss: 0.3812
Step 9000 (1.1 s/step), loss: 0.3768
Step 9100 (1.1 s/step), loss: 0.3865
Step 9200 (1.1 s/step), loss: 0.3886
Step 9300 (1.1 s/step), loss: 0.3841
Step 9400 (1.1 s/step), loss: 0.3821
Step 9500 (1.1 s/step), loss: 0.3783
Step 9600 (1.1 s/step), loss: 0.3797
Step 9700 (1.1 s/step), loss: 0.3793
Step 9800 (1.1 s/step), loss: 0.3793
Step 9900 (1.1 s/step), loss: 0.3747
Step 10000 (1.1 s/step), loss: 0.3914
/root/autodl-tmp/conda/envs/opendwm/lib/python3.10/site-packages/torch/distributed/fsdp/_state_dict_utils.py:768: UserWarning: When using ``NO_SHARD`` for ``ShardingStrategy``, full_state_dict willbe returned.
  warnings.warn(
/root/autodl-tmp/conda/envs/opendwm/lib/python3.10/site-packages/torch/distributed/fsdp/_state_dict_utils.py:711: UserWarning: When using ``NO_SHARD`` for ``ShardingStrategy``, full_state_dict willbe returned.
  warnings.warn(
[rank0]:W0328 17:19:55.303000 1251 site-packages/torch/distributed/distributed_c10d.py:2632] _object_to_tensor size: 70704 hash value: 10063483809014738815

I tried to train ctsd21 in nuscenes. When the 10000th steps, the message "_object_to_tensor size: 70704 hash value: 10063483809014738815" popped up, and the process got stuck. What's the problem with this?

**Comment by wzhgba** [comment]
If you train the pipeline with single GPU, all the FSDP related config items should be removed ([refer to 2.i 2.ii](https://github.com/SenseTime-FVG/OpenDWM/issues/17#issuecomment-2735175987)), otherwise the checkpoint saving (especially the optimizer state saving) will break.

**Comment by liaozhihao2757** [comment]
Thanks very much！It's work！I followed your advice and it worked fine with a single GPU. 

But when it came to the evaluation, it stayed for a very long time without returning results. There were 2082 items in the evaluation phase, so why did it take long? Is there any way to accelerate the evaluation?

**Comment by wzhgba** [comment]
Evaluation over the whole set takes long time, so usually run with distributed framework, or you can take a subset to evaluate by increasing the [stride](https://github.com/SenseTime-FVG/OpenDWM/blob/main/configs/ctsd/single_dataset/ctsd_21_crossview_tirda_bm_nusc_a.json#L408) of validation dataset
